hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java

```diff
@@ -579,11 +571,11 @@ private static void invokeSetStoragePolicy(final FileSystem fs, final Path path,
       toThrow = e;
       final String msg = "FileSystem doesn't support setStoragePolicy; HDFS-6584, HDFS-9345 " +
           "not available. This is normal and expected on earlier Hadoop versions.";
+
+      LOG.debug(msg, e);
```

Comment:
@busbey The difference here is that I may not want a WARN message every time, but I do want a DEBUG message.  Trying to add logic that says "WARN in these scenarios and DEBUG in these other scenarios" are not awesome because there will never be a unit test to validate and maintain that behavior.  Best to just always DEBUG and KISS.

Reply:
we have 0 log message unit tests. are we planning to add some? or am I missing some other context here?

the intention on the current state of "log once at WARN per classloader" is that these message are important for an operator but we don't want to flood the logs. The DEBUG level is available in case the system has been running long enough that we would have rolled off the WARN message and we need a way to elicit a temporary repeat with a runtime change.

Reply:
@busbey Not I did not mean to say that we need unit tests. I'm saying that it gets a bit tricky when there starts to be logic controlling DEBUG messages.  I think it's best to always record the state of the system in the DEBUG logging, and include WARN messages for only certain states. But I will put this back for you.